#! /usr/bin/env python

PACKAGE='starmac_robots_asctec'
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

STOP = SensorLevels.RECONFIGURE_STOP
gen = ParameterGenerator()
# Params:   Name  Type Reconfiguration level  Description  Default Min Max
gen.add("yaw_KP", double_t, STOP, "Yaw controller proportional gain [deg/s per deg]", 2.0, 0.0, 50.0)
gen.add("yaw_KI", double_t, STOP, "Yaw controller integral gain [deg/s per deg-second]", 0.1, 0.0, 50.0)
gen.add("yaw_KD", double_t, STOP, "Yaw controller derivative gain [deg/s per deg/s]", 0.0, 0.0, 50.0)
gen.add("yaw_Ilimit", double_t, STOP, "Yaw controller integral term limit [deg/s]", 5.0, 0.0, 50.0)
gen.add("roll_trim", double_t, STOP, "Trim value to be added to (body) roll command [deg]", 0.0, -10.0, 10.0)
gen.add("pitch_trim", double_t, STOP, "Trim value to be added to (body) roll command [deg]", 0.0, -10.0, 10.0)

exit(gen.generate(PACKAGE, "starmac_asctec_adapter", "AscTecAdapter"))